// components/basketcard/index.js
import React, { Component } from 'react'
import { Link } from 'react-router-3'
import { connect } from 'react-redux'
import Waypoint from 'react-waypoint'
import classNames from 'classnames'

import {
    getTotalBasketCount,
    getTotalBasketPrice
} from 'selectors'

class BasketCard extends Component {
    constructor(props) {
        super(props)
        this.state = {
            onBasket: true
        }
    }


    

    render() {
        const { totalBasketCount, totalPrice } = this.props
        // const linkClass =  classNames({
        //     'btn btn-inverse btn-block btn-lg': true,
        //     'btn btn-inverse-absolute btn-block btn-lg': !this.state.onBasket
        // })
        return (
            <Waypoint 
                onLeave={()=>{this.setState({onBasket: false})}}
            >
                <div className='cart'>
                    <div className='dropdown'>
                        <Link
                            to='/basket'
                            id='dLabel'
                            className='btn btn-inverse btn-block btn-lg'>
                            <i className='fa fa-fa-shoping-cart'></i>
                            <span>{totalBasketCount} item(s) - ${totalPrice}</span>
                        </Link>
                    </div>
                </div>
            </Waypoint>
        )
    }
}

// const BasketCard = ({ totalBasketCount, totalPrice }) => (

//         <div className='cart'>
//             <div className='dropdown'>
//                 <Link
//                     to='/basket'
//                     id='dLabel'
//                     className='btn btn-inverse btn-block btn-lg'>
//                     <i className='fa fa-fa-shoping-cart'></i>
//                     <span>{totalBasketCount} item(s) - ${totalPrice}</span>
//                 </Link>
//             </div>
//         </div>
// )

const mapStateToProps = state => ({
    totalBasketCount: getTotalBasketCount(state),
    totalPrice: getTotalBasketPrice(state)
})

export default connect(mapStateToProps, null)(BasketCard)
