// containers/phone/index.js
import React, { Component } from 'react'
import { connect } from 'react-redux'
import * as R from 'ramda'
import {Link} from 'react-router-3'

import { fetchPhoneById, addPhoneToBasket } from 'actions'
import { getphoneById } from 'selectors'
import BasketCard  from 'components/basketcard'


class Phone extends Component {
    componentDidMount() {
        this.props.fetchPhoneById(this.props.params.id)
    }
    renderFields() {
        const { phone } = this.props
        const columnFields = R.compose(
            R.toPairs,
            R.pick([
                'cpu',
                'size',
                'weight',
                'dispaly',
                'battary',
                'memory'
            ])
        )(phone)

        return columnFields.map(([key, value]) => (
            <div className='column' key={key}>
                <div className='ab-details-title'>
                    <p>{key}</p>
                </div>
                <div className='ab-details-info'>
                    {value}
                </div>
            </div>
        ))
    }

    renderContent() {
        const { phone } = this.props
        return <div className='thumbnail'>
            <div className='row'>
                <div className='col-md-6'>
                    <img className='img-thumbnail'
                        src={phone.image}
                        alt={phone.name} />
                </div>
                <div className='col-md-6'>
                    {this.renderFields()}
                </div>
            </div>
            <div className='caption-full'>
                <h4 className='pull-right'>
                    ${phone.price}
                </h4>
                <h4>{phone.name}</h4>
                <p>{phone.description}</p>
            </div>
        </div>
    }
    renderSidebar() {
        const { phone, addPhoneToBasket } = this.props
        return (
            <div>
                <p className='lead'>Quick shop</p>
                <BasketCard />
                <div className='form-group'>
                    <h1>{phone.name}</h1>
                    <h2>${phone.price}</h2>
                </div>
                <Link to='/' className='btn btn-info btn-block'>
                    Back To Store
                </Link>   
                <button 
                    type='button'
                    className='btn btn-success btn-block'
                    onClick={()=> addPhoneToBasket(phone.id)}
                >
                Add To Card
                </button> 
            </div>
        )
    }

    render() {
        console.log('phone', this.props.phone);
        
        
        return (
            <div className='view-container'>
                <div className='container'>
                    <div className='row'>
                        <div className='col-md-9'>
                            {this.props.phone && this.renderContent()}
                        </div>
                        <div className='col-md-3'>
                            {this.props.phone && this.renderSidebar()}
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}
const mapStateToProps = state => ({
    phone: getphoneById(state, state.phonePage.id)
})

const mapDispatchToProps = {
    fetchPhoneById,
    addPhoneToBasket
}

export default connect(mapStateToProps, mapDispatchToProps)(Phone)
