// actions/index.js
import {
    FETCH_PHONES_START,
    FETCH_PHONES_SUCCESS,
    FETCH_PHONES_FAILURE,
    LOAD_MORE_PHONES_START,
    LOAD_MORE_PHONES_SUCCESS,
    LOAD_MORE_PHONES_FAILURE,
    FETCH_PHONE_BY_ID_START,
    FETCH_PHONE_BY_ID_SUCCESS,
    FETCH_PHONE_BY_ID_FAILURE,
    ADD_PHONE_TO_BASKET,
    SERACH_PHONE,
    FETCH_CATIGORIES_START,
    FETCH_CATIGORIES_SUCCESS,
    FETCH_CATIGORIES_FAILURE,
    REMOVE_FROM_BASKET,
    CLEAR_BASKET

} from 'actionTypes'
import {fetchPhones as fetchPhonesApi,
        loadMorePhones as loadMorePhonesApi,
        fetchPhoneById as fetchPhoneByIdApi,
        fetchCatigories as fetchCatigoriesApi,
        fetchCat
    } from 'api'
import {getRenderedPhonesLength} from 'selectors'

export const fetchPhones = () => async dispatch => {
    
    dispatch({type: FETCH_PHONES_START})

    try {
        const phones = await fetchPhonesApi()
        // const phonesAll = await fetchPhonesApi(0, 13)
        dispatch({
            type: FETCH_PHONES_SUCCESS,
            payload: phones
        })
        
    } catch (err) {
       dispatch({
           type: FETCH_PHONES_FAILURE,
           payload: err,
           error: true
       }) 
    }
}

export const loadMorePhones = () => async (dispatch, getState) => {
    const offset = getRenderedPhonesLength(getState())
    dispatch({type: LOAD_MORE_PHONES_START})

    try {
        const phones = await loadMorePhonesApi({offset})
        dispatch({
            type: LOAD_MORE_PHONES_SUCCESS,
            payload: phones
        })
    } catch (err) {
       dispatch({
           type: LOAD_MORE_PHONES_FAILURE,
           payload: err,
           error: true
       }) 
    }
}

export const fetchPhoneById = id => async dispatch => {
    dispatch({type: FETCH_PHONE_BY_ID_START})
    

    try {
        const phone = await fetchPhoneByIdApi(id)
       dispatch({
        type: FETCH_PHONE_BY_ID_SUCCESS,
        payload: phone
       })
    } catch(err) {
       dispatch({
        type: FETCH_PHONE_BY_ID_FAILURE,
        payload: err,
        error: true 
       })
    }
}

export const addPhoneToBasket = id => dispatch => {
    dispatch({
        type: ADD_PHONE_TO_BASKET,
        payload: id
    })
}

export const searchPhone = text => dispatch => {
    dispatch({
        type: SERACH_PHONE,
        payload: text
    })
}

export const fetchCatigories = () => async dispatch => {
    dispatch({type: FETCH_CATIGORIES_START})

    try {
        const catigories = await fetchCatigoriesApi()
        dispatch({
            type: FETCH_CATIGORIES_SUCCESS,
            payload: catigories
        })
    } catch (err) {
       dispatch({
           type: FETCH_CATIGORIES_FAILURE,
           payload: err,
           error: true
       }) 
    }
}

export const removePhoneFromBasket = id => dispatch => {
    dispatch({
        type: REMOVE_FROM_BASKET,
        payload: id
    })
}

export const cleanBasket = () => dispatch => {
    dispatch({
        type: CLEAR_BASKET
    })
}

export const basketCheckout = phones => () => {
    alert(JSON.stringify(phones))
}
